### PR TITLE
move ndarray methods to free functions

### PR DESCRIPTION
**Pull request by ev-br**
closes gh-60 when done

### CODE REVIEW

**Comment by ev-br** [comment]
No longer WIP. This a very boring PR, mostly just shifting around the boilerplate. And several ways of cutting this boilerplate are obvious. The plan is to level the field before addressing gh-32.

File: torch_np/_detail/implementations.py [hunk 1]
```diff
@@ -624,6 +646,15 @@ def round(tensor, decimals=0):
     return result
 
 
+def imag(tensor):
+    try:
+        result = tensor.imag
+    except RuntimeError:
+        # RuntimeError: imag is not implemented for tensors with non-complex dtypes.
+        result = torch.zeros_like(tensor)
```
Comment (lezcano) [inline torch_np/_detail/implementations.py:654]: Just do an if-else on the type of the tensor to be more explicit.

**Comment by ev-br** [comment]
NB https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/70 follows up on this.

**Comment by ev-br** [comment]
Thanks Mario for the review.

**Pull request merged**